Scripts/Runtime/Entities/ProxyDataStream/ProxyDataStreamExtension.cs

```diff
@@ -0,0 +1,46 @@
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Helper methods when working with <see cref="ProxyDataStream{TData}"/>
+    /// These make it more clear what is happening when operating on <see cref="ProxyDataStream{TData}"/> instances
+    /// in a job.
+    /// </summary>
+    public static class ProxyDataStreamExtension
+    {
+        //TODO: RE-ENABLE IF NEEDED
+        //TODO: DOCS
+        // public static void Resolve<TData, TResult, TEnum>(this TData data, TEnum option, ref TResult result, ref PDSUpdater<TData> updater)
+        //     where TData : unmanaged, IProxyData
+        //     where TResult : unmanaged, IProxyData
+        //     where TEnum : unmanaged, Enum
+        // {
+        //     VDResultsDestination<TResult> resultsDestination = data.ResultsDestinationLookup.GetVDResultsDestination<TEnum, TResult>(option);
+        //     VDResultsWriter<TResult> resultsWriter = resultsDestination.AsResultsWriter(updater.CurrentContext);
+        //     resultsWriter.Add(ref result, updater.LaneIndex);
+        //     updater.Resolve();
+        // }
+        //
+        // //TODO: DOCS
+        // public static void Resolve<TData, TResult, TEnum>(this TData data, TEnum option, TResult result, ref PDSUpdater<TData> updater)
+        //     where TData : unmanaged, IProxyData
+        //     where TResult : unmanaged, IProxyData
+        //     where TEnum : unmanaged, Enum
+        // {
+        //     Resolve(data, option, ref result, ref updater);
+        // }
+
+        //TODO: DOCS
+        public static void Resolve<TData>(this TData data, ref PDSUpdater<TData> updater)
+            where TData : unmanaged, IProxyData
+        {
+            updater.Resolve();
```

Comment:
This will change a bit with the new architecture but previously how this worked was that the `PDSUpdater` would give you an instance of data. And the `PDSUpdater` would enter a `Modifying` state. 

Then you would either have to call `ContinueOn` or `Resolve` in your job. 

This would flip the `PDSUpdater` back to a `Ready` state. If you tried to modify other data without resolving or continuing an exception would be thrown. 

This specific `Resolve` extension function was for the case where an instance of data doesn't have a result, it just shouldn't exist anymore so it gets Resolved to nothing. We don't care about the data, we just use it to trigger the extension method and the updater resolves.

Reply:
That's fair, I'll add a TODO to think about this when I re-enable. 

Originally, we had felt it was annoying to have inconsistency in the API.

```
myData.Resolve(new Result(), ref m_Updater);

m_Updater.ResolveWithNoResult();
```